from snakemake.utils import min_version
min_version("6.15")


configfile: "config/config.yaml"


module univec_removal:
    snakefile: "../workflow/Snakefile"
    config: config


use rule * from univec_removal


rule test_all:
    input:
        "diff.txt",
        expand("results/trimmed/UniVec{core}_filtered/test/1.{read}/fastqc_report.html",
            core="_Core" if config["univec"]["core"] else "",
            read=["1", "2"],
        ),
    default_target: True


rule test_output:
    input:
        processed=expand("results/trimmed/UniVec{core}_filtered/test/1.{read_number}.fastq.gz",
            core="_Core" if config["univec"]["core"] else "",
            read_number=["1", "2"],
        ),
        expected=expand("data/mason_genome.contig-length_6500.seed_37.{read_num}.fq",
            read_num=["R1", "R2"],
        ),
    output:
        "diff.txt",
    shell:
        """
        diff <( zcat {input.processed} ) <( cat {input.expected} | sed -r -e 's#(@mason_genome\\.contig-length_6500\\.seed_37_[0-9]+)/[12]#\\1#' ) >{output} 2>&1
        """
